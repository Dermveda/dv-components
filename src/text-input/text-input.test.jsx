import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import TextInput from './text-input.jsx';
const question = {
	_id: 'abc123',
	name: 'test q',
	type: 'test',
	isCorrect: true,
	answers: [
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		}
	]
};

describe('<TextInput/>', () => {
	it('Will render', () => {
		const el = renderer.create(<TextInput question={question} />);
		expect(el).toMatchSnapshot();
	});
	it('Will select answer', () => {
		const onChangeFunc = jest.fn();
		const el = mount(<TextInput question={question} onSelectAnswer={onChangeFunc} />);
		el.find('input').simulate('change');
		expect(onChangeFunc).toHaveBeenCalledTimes(1);
	});
});
