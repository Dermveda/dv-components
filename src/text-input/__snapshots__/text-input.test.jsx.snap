// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> Will render 1`] = `
<li>
  test q
  <div
    className="form-group dvc-form-group"
  >
    <label
      htmlFor="TextInput"
    >
      
    </label>
    <input
      autoComplete="off"
      className="form-control "
      disabled={undefined}
      id="TextInput"
      name="TextInput"
      onChange={[Function]}
      placeholder="Enter response here."
      required={undefined}
      type="text"
      value={undefined}
    />
    <div
      className="invalid-feedback"
    >
      Please fill in field.
    </div>
    <small
      className="form-text text-muted"
    />
  </div>
</li>
`;
